plasmapy/tests/test_requirements.py

```diff
@@ -0,0 +1,141 @@
+"""Tests for the consistency of requirements."""
+
+import os
+import pytest
+import setuptools
+import tomli
+
+from typing import Dict, Set
+
+import plasmapy
+
+base_directory = os.path.realpath(f"{plasmapy.__path__[0]}/..")
+requirements_directory = f"{base_directory}/requirements"
+requirements_prefixes = ("build", "docs", "extras", "install", "tests")
+
+
+def read_requirements_txt_file(prefix: str, requirements_directory: str) -> Set[str]:
+    """
+    Read in a text file containing requirements.
+
+    Parameters
+    ----------
+    prefix : str
+        The prefix to a filename, such as ``"build"`` for ``build.txt``.
+
+    requirements_directory : str
+        The path to the directory containing the requirements file.
+
+    Returns
+    -------
+    list of str
+        A `list` containing the lines of the requirements file,
+        excluding lines that do not start with an alphabetic character
+        (like comments).
+    """
+    filename = f"{requirements_directory}/{prefix}.txt"
+    with open(filename) as file:
+        lines_of_file = file.readlines()
+    return {line.strip() for line in lines_of_file if line[0].isalpha()}
+
+
+def get_requirements_from_txt() -> Dict[str, Set[str]]:
+    """
+    Get the requirements from the ``.txt`` files in the ``requirements``
+    directory.
+    """
+    return {
+        prefix: read_requirements_txt_file(prefix, requirements_directory)
+        for prefix in requirements_prefixes
+    }
```

Comment:
A point on design. I've been reading https://web.stanford.edu/~ouster/cgi-bin/book.php lately, and I can't help but notice that these few functions you've got here are rather shallow - they don't hide much complexity, if any. I get the fear of long functions, but I think these oneliners could be, well, inlined.